Sent upstream.
WasmSignalHandlers.cpp should use __sun instead of SOLARIS
https://bugzilla.mozilla.org/show_bug.cgi?id=1353050

--- firefox-52.0b6/js/src/wasm/WasmSignalHandlers.cpp
+++ firefox-52.0b6/js/src/wasm/WasmSignalHandlers.cpp
@@ -118,7 +118,7 @@
 # define R13_sig(p) ((p)->sc_r13)
 # define R14_sig(p) ((p)->sc_r14)
 # define R15_sig(p) ((p)->sc_r15)
-#elif defined(__linux__) || defined(SOLARIS)
+#elif defined(__linux__) || defined(__sun)
 # if defined(__linux__)
 #  define XMM_sig(p,i) ((p)->uc_mcontext.fpregs->_xmm[i])
 #  define EIP_sig(p) ((p)->uc_mcontext.gregs[REG_EIP])
